hivemind/optim/experimental/optimizer.py

```diff
@@ -31,91 +31,100 @@
+    >>> opt = hivemind.Optimizer(dht=dht, run_id="run_42", params=model.parameters(),
+    >>>                          optimizer=lambda params: torch.optim.Adam(params, **other_options),
+    >>>                          target_batch_size=4096, batch_size_per_step=4)
     >>> while True:
     >>>     loss = compute_loss_on_batch(model, batch_size=4)
     >>>     opt.zero_grad()
     >>>     loss.backward()
     >>>     opt.step()  # <-- train collaboratively with any peers that use the same prefix (run_42)
 
-    However, unlike regular optimizers, calling opt.step with hivemind.Optimizer can do one of the following:
-
-     - accumulate a minibatch of gradients towards the (global) target batch size, without updating parameters yet;
-     - after accumulating the target batch size, all-reduce gradients with peers and perform optimizer step;
-     - if your peer lags behind the rest of the swarm, it will download latest state from other peers;
-
-    :example: the optimizer has many keyword arguments that may be difficult to understand in one go. Here's quickstart
-      that will help you setup your first synchronous optimizer.
-
-    >>> hivemind.Optimizer(
-    >>>    dht=hivemind.DHT(initial_peers=ADDRESS_HERE, client_mode=TRUE_IF_BEHIND_FIREWALL_OR_UNRELIABLE, start=True),
-    >>>    run_id="a_unique_name_that_every_participant_will_see_when_training",
-    >>>    batch_size_per_step=ACTUAL_BATCH_SIZE_OF_THIS_PEER,
-    >>>    target_batch_size=LARGE_GLOBAL_BATCH,  # global batch will be this or *slightly* larger due to stragglers;
-    >>>      #  peers should finish averaging in roughly half the time they need to accumulate this batch between them
-    >>>    optimizer=lambda params: AnyPyTorchOptimizer(params, **config_that_makes_sense_for_target_batch_size),
-    >>>      # ^-- scale learning rate for your target_batch_size; good reference: https://arxiv.org/abs/1904.00962
-    >>>    offload_optimizer=True,  # this saves GPU memory; large-batch training does not need optimizer that often
-    >>>    scheduler=lambda opt: AnyPytTorchScheduler(opt, **config_that_makes_sense_for_target_batch_size),
-    >>>      # scheduler.step will be called once every time peers collectively accumulate target_batch_size
-    >>>    matchmaking_time=15.0, averaging_timeout=60.0,  # <-- if the network is fast reduce to 3-5s and 10-15s
-    >>>      # increase matchmaking_time if at least 25% of the time you see "averaged gradients with <...> peers",
-    >>>      # ... but N is less than 0.9x the actual number of peers. Increase averaging_timeout if half of the epochs
-    >>>      # ... print "Proceeding with local gradients" instead of "Averaged gradients with N peers"
+    By default, peers will perform the following steps:
+
+     * accumulate a minibatch of gradients towards the (global) target batch size, without updating parameters yet;
```

Comment:
The bulletpoints are not actually displayed in this and other lists:

<img width="394" alt="Screenshot 2021-11-30 at 15 45 08" src="https://user-images.githubusercontent.com/8748943/144058547-545fb7cd-3e12-4f91-92ab-7490e2e46205.png">

Reply:
Okay, it would appear that docutils==0.16 does not have this issue